#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: various documentation fix
# DP:	argp/argp.h: Bug#117680 patched by H. S. Teoh.
# DP:	manual/terminal.texi: Bug#133336 patched by GOTO Masanori.
# DP: Author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Not submitted
# DP: Status Details: Plan to submit if they are gathered.
# DP: Date: 2002-12-31

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- argp/argp.h.ORIG	2002-12-30 09:08:17.000000000 -0500
+++ argp/argp.h	2002-12-30 09:10:28.000000000 -0500
@@ -96,7 +96,7 @@
    is set, then the option NAME field is displayed unmodified (e.g., no `--'
    prefix is added) at the left-margin (where a *short* option would normally
    be displayed), and the documentation string in the normal place.  For
-   purposes of sorting, any leading whitespace and puncuation is ignored,
+   purposes of sorting, any leading whitespace and punctuation is ignored,
    except that if the first non-whitespace character is not `-', this entry
    is displayed after all options (and OPTION_DOC entries with a leading `-')
    in the same group.  */
--- manual/terminal.texi	2002-10-11 02:50:16.000000000 +0900
+++ manual/terminal.texi.new	2002-12-29 12:16:50.000000000 +0900
@@ -1617,7 +1617,7 @@
 
 @comment termios.h
 @comment BSD
-@deftypefun int cfmakeraw (struct termios *@var{termios-p})
+@deftypefun void cfmakeraw (struct termios *@var{termios-p})
 This function provides an easy way to set up @code{*@var{termios-p}} for
 what has traditionally been called ``raw mode'' in BSD.  This uses
 noncanonical input, and turns off most processing to give an unmodified
